test/wallet.test.ts

```diff
@@ -223,6 +272,37 @@ describe('wallet', () => {
         );
       }
     });
+
+    it('should throw with unknown address', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice();
+      const witnessedMsgParams: MessageParams[] = [];
+      const processTypedMessage = async (msgParams: MessageParams) => {
+        witnessedMsgParams.push(msgParams);
+        return testMsgSig;
+      };
+
+      engine.push(createWalletMiddleware({ getAccounts, processTypedMessage }));
+      const message = [
+        {
+          type: 'string',
+          name: 'message',
+          value: 'Hi, Alice!',
+        },
+      ];
+
+      const payload = {
+        method: 'eth_signTypedData',
+        params: [message, testUnkownAddress],
+      };
+      try {
+        await pify(engine.handle).call(engine, payload);
+      } catch (e: any) {
+        expect(e.message).toEqual(
+          'The requested account and/or method has not been authorized by the user.',
+        );
+      }
```

Comment:
Similar to another comment, it's better to write this as the following to rule out a false positive:

``` suggestion
      const promise = pify(engine.handle).call(engine, payload);
      await expect(promise).rejects.toThrow('The requested account and/or method has not been authorized by the user.')
```